assets/tests/hooks/useDetour.test.ts

```diff
@@ -149,70 +81,11 @@ describe("useDetour", () => {
 
     await waitFor(() => {
       expect(result.current.detourShape).toStrictEqual([])
-      expect(result.current.directions).toStrictEqual(undefined)
       expect(result.current.routingError).toBeTruthy()
     })
   })
 
-  test("when `undo` is called, removes `start` and `end` points", async () => {
-    const start = { lat: 0, lon: 0 }
-    const end = { lat: 1, lon: 1 }
-
-    const { result } = renderHook(useDetourWithFakeRoutePattern)
-
-    act(() => result.current.addConnectionPoint?.(start))
-    act(() => result.current.addConnectionPoint?.(end))
-
-    expect(result.current.endPoint).not.toBeNull()
-
-    act(() => result.current.undo?.())
-
-    expect(result.current.endPoint).toBeNull()
-    expect(result.current.startPoint).not.toBeNull()
-
-    act(() => result.current.undo?.())
-
-    await waitFor(() => expect(result.current.startPoint).toBeNull())
-  })
-
-  test("when `undo` is called, removes the last `waypoint`", async () => {
-    const start = { lat: 0, lon: 0 }
-    const end = { lat: 1, lon: 1 }
-
-    const { result } = renderHook(useDetourWithFakeRoutePattern)
-
-    act(() => result.current.addConnectionPoint?.(start))
-    act(() => result.current.addWaypoint?.(end))
-
-    expect(result.current.waypoints).toStrictEqual([end])
-
-    act(() => result.current.undo?.())
-
-    await waitFor(() => expect(result.current.waypoints).toHaveLength(0))
-  })
-
-  test("when `undo` is called, should call API with updated waypoints", async () => {
-    const start = { lat: 0, lon: 0 }
-    const mid = { lat: 0.5, lon: 0.5 }
-    const end = { lat: 1, lon: 1 }
-
-    const { result } = renderHook(useDetourWithFakeRoutePattern)
-
-    act(() => result.current.addConnectionPoint?.(start))
-    act(() => result.current.addWaypoint?.(mid))
-    act(() => result.current.addWaypoint?.(end))
-    act(() => result.current.undo?.())
-
-    await waitFor(() => {
-      expect(jest.mocked(fetchDetourDirections)).toHaveBeenCalledTimes(3)
-      expect(jest.mocked(fetchDetourDirections)).toHaveBeenNthCalledWith(3, [
-        start,
-        mid,
-      ])
-    })
-  })
-
-  test("when `undo` removes the last waypoint, `detourShape` and `directions` should be empty", async () => {
+  test("clears `detourShape` when `undo` removes the last waypoint", async () => {
```

Comment:
This test case got moved to `"clicking on 'Undo' restores the placeholder text when appropriate"` in `diversionPage.test.tsx`.

I didn't move assertions about `routeSegments` or `detourShape` because I couldn't figure out how to make those assertions at the `diversionPage` level. I think it's _gotta_ be possible, but I spun my wheels for a bit and decided that it was a problem for a different day.

https://github.com/mbta/skate/pull/2580/files#diff-1535a72a38a9500ad346d1b3d9735aff2c788a2d97cfbf49f2aeaac7aefb26cfR366-R399

Reply:
I can imagine a regression where while the placeholder text is restored when "undo" is clicked, the directions are not removed as expected. Does that seem to warrant a check here?

Reply:
Yeah probably. Will add

Reply:
The fact that there's no way to trigger that without contorting the code is itself an implementation detail, so yes, this should be added.

Reply:
[Feedback: Validate that directions are removed when placeholder text is visible](https://github.com/mbta/skate/pull/2580/commits/1810874fbbb8c2f906664c60cf13b54064e92e41)